### ISSUE TITLE
lam is computed incorrectly in RandomCutMixV2

### ISSUE THREAD

**Issue by m1pro** [kornia/kornia#3077]
### Describe the bug

The computation of lambda in this line:
https://github.com/kornia/kornia/blob/52cee42a15f4828adf729e28931479ec5375ed20/kornia/augmentation/_2d/mix/cutmix.py#L90

should be:
```python
lam = 1 - w.to(input.dtype) * h.to(input.dtype) / (width * height)  # width_beta * height_beta 
```

If aligning to the way the loss is computed in the paper, i.e lambda multiplies the loss that is computed with the first label:
![image](https://github.com/user-attachments/assets/b21f6296-141f-4b6e-8265-ef58e88005b8)

lamba  should be computed as in the paper:
![image](https://github.com/user-attachments/assets/317a87f7-14cc-45db-9ceb-2dc22ef8ce7d)

**Comment by edgarriba** [comment]
@shijianjian did you consider that or is it definitely a bug ?

**Comment by m1pro** [comment]
If aligning to the way the loss is computed in the paper, i.e lambda multiplies the loss that is computed with the first label:
![image](https://github.com/user-attachments/assets/b21f6296-141f-4b6e-8265-ef58e88005b8)

 then yes, it should be computed as in the paper:
![image](https://github.com/user-attachments/assets/317a87f7-14cc-45db-9ceb-2dc22ef8ce7d)

**Comment by shijianjian** [comment]
@m1pro what should be the fix you think? I am a bit worried regarding the backward compatibility if we change it directly.

**Comment by m1pro** [comment]
I agree with the backward compatibility concern, but I think consistency of the code with the paper (and other implementations - sometimes one wants to switch to a different implementation for various reasons and that yields the need to change the signs in the computation of the loss) is more important.

Maybe, you can add a flag argument as input to the RandomCutMixV2 that controls the flipping of the sign and is set by default for now to False (i.e. keeping current behavior) and at the same time prints a deprecation warning message for a transition period. Users who want to work with the correct sign can do so by setting this flag to True.

Later on after an adaptation period, it will be possible to remove the flag entirely and keep the correct sign.

**Comment by shijianjian** [comment]
OK. Sounds good. Can you raise a PR for this?
